### PR TITLE
feat(installer): add `lvim` and `lvim.cmd` for windows

### PR DESCRIPTION
**Pull request by younger-1**
# Description

Add two new shims for windows: `lvim` and `lvim.cmd`

`lvim` is used by `git.exe` to take as core editor when commit message or other git editing tasks.

`lvim.cmd` is used in `cmd.exe` or other situations which `PowerShell` is not convenient.

### CODE REVIEW

**Review by kylo252 — COMMENTED** [review]
I still don't why we need `lvim.cmd`, can you give a full example workflow? 
~~I have a feeling this might be easily solved by using `git config`.~~ Okay, this actually took more effort than I'd like to admit. The problem is that `git` was obnoxiously picky about slashes, but this should work:

```powershell
git config --global core.editor "powershell.exe c:\\Users\\$env:USERNAME\\.local\\bin\\lvim.ps1"
```

If it works for you and you don't need that `lvim.cmd` anymore, then we can take the opportunity to either alter this PR or create a new one that includes a windows-specific version of https://github.com/LunarVim/LunarVim/blob/rolling/utils/installer/config.example.lua, with some tweaks to get people up and running more quickly.

**Comment by kylo252** [comment]
See https://github.com/LunarVim/LunarVim/issues/1902 for more info with regards to a new `config.example_windows.lua`

File: utils/installer/install.ps1 [hunk 1]
```diff
@@ -205,10 +205,29 @@ function clone_lvim() {
...
+
+    # 1. lvim.ps1
+    if ((Test-Path "$INSTALL_BIN_DIR") -eq $false) {
+        New-Item "$INSTALL_BIN_DIR" -ItemType Directory
     }
-    Copy-Item "$env:LUNARVIM_RUNTIME_DIR\lvim\utils\bin\lvim.ps1" -Destination "$INSTALL_PREFIX\bin\lvim.ps1" -Force
+    Copy-Item "$lvim_ps1_path" -Destination "$INSTALL_BIN_DIR\lvim.ps1" -Force
+
+    # 2. lvim: for git.exe use lvim as core editor
+    Set-Content -Path $INSTALL_BIN_DIR\lvim -Value $(-join @("#!/bin/sh", "`r`n", ('powershell.exe -noprofile -ex unrestricted "{0}" "$@"' -f $lvim_ps1_path))) -Encoding ascii
```
[the remark this answers is not in the document]
Comment (younger-1) [reply utils/installer/install.ps1:217]: Oh, your trick is good. I am willing to use cleaner way and I just want to share the same `gitconfig` file between windows and linux. 😃

File: utils/installer/install.ps1 [hunk 2]
```diff
@@ -205,10 +205,29 @@ function clone_lvim() {
...
+setlocal enabledelayedexpansion
+set args=%*
+:: replace problem characters in arguments
+set args=%args:`"='%
+set args=%args:(=``(%
+set args=%args:)=``)%
+set invalid=`"='
+if !args! == !invalid! ( set args= )
+powershell -noprofile -ex unrestricted `"& '$lvim_ps1_path' $arg %args%;exit `$lastexitcode`"" 
+    $content | Out-File $INSTALL_BIN_DIR\lvim.cmd -Encoding ascii
```
[the remark this answers is not in the document]
Comment (younger-1) [reply utils/installer/install.ps1:230]: `lvim.cmd` is not so useful except that powershell is not required to open neovim.

It is possible to open lvim in `nvim-qt` (or `neovide` , `fvim`) in this manner:
```
# in cmd
lvim --headless --listen localhost:6666
nvim-qt.exe --server=localhost:6666
```

**Comment by younger-1** [comment]
Another use case of `lvim.cmd` is to launch neovim GUI client.
As far as I know, if I want to launch `neovide` with `lunarvim` inside, I have to pass in `lvim.cmd` and `lvim.ps1` didn't work

```
$env:NEOVIM_BIN = "C:\Users\younger\.local\bin\lvim.cmd"
neovide.exe
```

**Comment by yatli** [comment]
It's also possible for FVim: `fvim --nvim lvim`

**Pull request closed without merging**